### DISCUSSION TITLE
How can I embed HTML images/files into my website?

### DISCUSSION THREAD

**Discussion by jeetyadav98** [squidfunk/mkdocs-material#6732] [Q&A]
Title self explanatory. I basically have a lot of interactive plots which I save in html, and would like to be able to embed and interact with them in my website as well. 

I've tried multiple ways - including it as a source file (which works for csv), using the html extrension etc. But none worked. 

Any idea how this can be achieved?

**Comment by kamilkrzyskow** [comment]
Hello @jeetyadav98,
for images a [`<img>`](https://www.w3schools.com/tags/tag_img.asp) tag should suffice, for HTML files an [`<iframe>`](https://www.w3schools.com/tags/tag_iframe.asp) tag would be the way to go. There is also [`<embed>`](https://www.w3schools.com/tags/tag_embed.asp) but in the "Suggestion" on the page other tags are preferred over it.

If you have tried an iframe already, and it didn't work, then be more specific about it what happened. The file didn't load, did you check the Developer Tools and saw a 404 error or everything loaded and the iframe was still empty etc.?

Note that an embedded file won't sync with your theme palette color etc. if you'd prefer for it to be synced, you'd need to extract the contents of the HTML's `<body>` tag and paste them into the Markdown\* and then adjust styling of the newly included tags with additional CSS. Also any JavaScript libraries that were included in the HTML's `<head>` would need to be included in the Markdown as well.
If there are not too many files like that you want to copy over, perhaps it would be "cleaner" to do via [theme extension](https://squidfunk.github.io/mkdocs-material/customization/#extending-the-theme). 

> \* The Python Markdown syntax is rather strict, so perhaps you'd need to keep the file structure flat without indentation for nested tags etc.

> Perhaps related discussions:
https://github.com/squidfunk/mkdocs-material/discussions/6658
https://github.com/squidfunk/mkdocs-material/discussions/6503

**Comment by ghost** [comment]
you could try the "svg" tag, here is the link to something about it i found on a website: https://www.w3schools.com/html/html5_svg.asp
